minimalTest.js

```diff
@@ -53,7 +52,10 @@ function setRoomList() {
         }
         return 0;
     });
-    roomList = roomList.reverse()
+    tmp = tmp.reverse()
+    let tmp2 = new Map();
+    tmp.map((room) => {tmp2.set(room.roomId, room)})
+    roomList = tmp2
```

Comment:
should probably leave a comment here that we're making use of `Map`'s insertion order sort which is why we need to create new map and reassign to `roomList` else we're gonna forget